### PR TITLE
Cr 1541 page titles and urls

### PR DESCRIPTION
**Pull request by Paul-Joel**
# Motivation and Context
Updates to RH page titles and urls

# What has changed
Update to change 'requests' to 'request' as user_journey and url part throughout
Change of 'start/save-and-exit' to 'signed-out', and update of value in EQ token generation
Change to error trapping to route error back to the 'GET' version of the page, to ensure that a page refresh clears the error
Updates to page title content and generation throughout
Change start/unlinked to start/link-address
Various minor text and class updates
Split of error 'address-not-found' into its own page
Removal of unlinked text and fix to error display in 'enter-address'
Unit test updates for changed urls and addition of test for new page titles

Requires Cucumber updates due to text change in enter-address
Requires terraform update for ingress url changes

# How to test?

# Links
https://collaborate2.ons.gov.uk/jira/browse/CR-1541
https://github.com/ONSdigital/census-rh-cucumber/pull/114
https://github.com/ONSdigital/census-rh-terraform/pull/318

### CODE REVIEW

**Review by cookel2 — APPROVED** [review]
looks good - I've made a refactoring suggestion to potentially improve efficiency at runtime

File: app/templates/partials/messages.html [hunk 1]
```diff
@@ -4,13 +4,13 @@
 {% from './components/lists/_macro.njk' import onsList %}
 
 {% if messages | length == 1 %}
-    {% if page_url == '/start/' %}
```
Comment (MartinHumphrey) [inline app/templates/partials/messages.html:7]: Previously wrong with only Start page having a difference but now Start journey is different to others? Seems strange having inconsistency on different journeys.

File: app/templates/partials/messages.html [hunk 2]
```diff
@@ -4,13 +4,13 @@
...
+    {% if '/start/' in page_url %}
         {% set errorTitle = _('There is a problem with this page') %}
     {% else %}
         {% set errorTitle = _('There is a problem with your answer') %}
     {% endif %}
 {% else %}
-    {% if page_url == '/start/' %}
+    {% if '/start/' in page_url %}
```
Comment (MartinHumphrey) [inline app/templates/partials/messages.html:13]: Why are journeys inconsistent?

File: app/request_handlers.py [hunk 3]
```diff
@@ -617,15 +571,11 @@ async def post(self, request):
         if not form_valid:
             logger.info('form submission error',
                         client_ip=request['client_ip'])
-            return {
-                'form_value_name_first_name': data.get('name_first_name'),
-                'form_value_name_last_name': data.get('name_last_name'),
```
Comment (MartinHumphrey) [inline app/request_handlers.py:622]: If errored 'First Name' or 'Last Name' are not passed back into the form anymore so for instance you get "Enter your last name" but your first name hasn't been saved if you entered it?

File: app/templates/request-common-enter-name.html [hunk 4]
```diff
@@ -44,7 +44,7 @@
                 'label': {
                     'text': _('First name')
                 },
-                'value': form_value_name_first_name,
+                'value': first_name,
```
Comment (MartinHumphrey) [inline app/templates/request-common-enter-name.html:47]: Value not available from context?

File: app/templates/request-common-enter-name.html [hunk 5]
```diff
@@ -56,7 +56,7 @@
                 'label': {
                     'text': _('Last name')
                 },
-                'value': form_value_name_last_name,
+                'value': last_name,
```
Comment (MartinHumphrey) [inline app/templates/request-common-enter-name.html:59]: Value not available from context?

File: app/web_form_handlers.py [hunk 6]
```diff
@@ -116,17 +113,8 @@ async def post(self, request):
 
         if not form_valid:
             logger.info('web form submission error', client_ip=request['client_ip'])
-            return {
-                'form_value_name': data.get('name'),
-                'form_value_country': data.get('country'),
-                'form_value_category': data.get('category'),
-                'form_value_email': data.get('email'),
-                'form_value_description': data.get('description'),
```
Comment (MartinHumphrey) [inline app/web_form_handlers.py:124]: On error none of the above values are now available to the form. The error page has the values which were not entered previously in the error panel, but the previously entered values have disappeared as well?

File: app/web_form_handlers.py [hunk 7]
```diff
@@ -116,17 +113,8 @@ async def post(self, request):
...
-                'display_region': display_region,
-                'page_title': page_title,
-                'locale': locale,
-                'page_url': View.gen_page_url(request)
-            }
+            raise HTTPFound(
+                request.app.router['WebForm:get'].url_for(display_region=display_region))
```
Comment (MartinHumphrey) [inline app/web_form_handlers.py:117]: If data error on POST why issue a redirect with a HTTP 302 and ask the browser to make another request. Why not just render the response with error messages in the reply?

File: app/webchat_handlers.py [hunk 8]
```diff
@@ -160,15 +164,8 @@ async def post(self, request):
         if not form_valid:
             logger.info('form submission error',
                         client_ip=request['client_ip'])
-            return {
-                'form_value_screen_name': data.get('screen_name'),
-                'form_value_country': data.get('country'),
-                'form_value_query': data.get('query'),
```
Comment (MartinHumphrey) [inline app/webchat_handlers.py:166]: On error none of the above values are now available to the form. The error page has the values which were not entered previously in the error panel, but the previously entered values have disappeared as well?

File: app/webchat_handlers.py [hunk 9]
```diff
@@ -160,15 +164,8 @@ async def post(self, request):
...
-                'display_region': display_region,
-                'page_title': page_title,
-                'locale': locale,
-                'page_url': View.gen_page_url(request)
-            }
+            raise HTTPFound(
+                request.app.router['WebChat:get'].url_for(display_region=display_region))
```
Comment (MartinHumphrey) [inline app/webchat_handlers.py:168]: If data error on POST why issue a redirect with a HTTP 302 and ask the browser to make another request. Why not just render the response with error messages in the reply?

File: tests/unit/__init__.py [hunk 10]
```diff
@@ -771,7 +790,7 @@ def setUp(self):
         }
 
         self.account_service_url = '/start/'
-        self.account_service_log_out_url = '/start/save-and-exit/'
+        self.account_service_log_out_url = '/signed-out/'
```
Comment (MartinHumphrey) [inline tests/unit/__init__.py:793]: account_service_log_out_url in eq_payload json, line 782 hasn't been updated as per this new log out url.

**Review by MartinHumphrey — CHANGES_REQUESTED** [review]
Mainly the same comment. Don't see why errors in a form submission should prompt a redirect to the same form rather than just render the page with the errors. Has also lead to problems with forms with multiple input items, the completed parts of the form are no longer sent back to the client for resubmission with the rest of the form completed.

**Pull request merged**